diff --git a/ios/FileSystemUtilities/FileSystemUtilities.swift b/ios/FileSystemUtilities/FileSystemUtilities.swift
index 6d3ad44994534e7fae84d6a8ea4606f4aedd0a77..eb8ed6f7c474d786558118f08601d8d7ddf5b50d 100644
--- a/ios/FileSystemUtilities/FileSystemUtilities.swift
+++ b/ios/FileSystemUtilities/FileSystemUtilities.swift
@@ -93,6 +93,17 @@ public struct FileSystemUtilities {
     return [.read, .write]
   }
 
+  public static func isReadableFile(_ appContext: AppContext?, _ url: URL) -> Bool {
+    let grantedPermissions = permissions(appContext, for: url)
+    guard grantedPermissions.contains(.read) else {
+      return false
+    }
+    guard url.isFileURL else {
+      return true
+    }
+    return FileManager.default.isReadableFile(atPath: url.path)
+  }
+
   private static func getAppGroupSharedDirectories(_ appContext: AppContext) -> [String] {
     let appGroups = appContext.appCodeSignEntitlements.appGroups ?? []
     var appGroupSharedDirectories: [String] = []
